scripts/quests/sandoria/Distant_Loyalties.lua

```diff
@@ -0,0 +1,161 @@
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Femitte'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if player:hasKeyItem(xi.ki.GOLDSMITHING_ORDER) then
+                        return quest:event(661)
+                    elseif
+                        questProgress == 3 and
+                        player:hasKeyItem(xi.ki.MYTHRIL_HEARTS)
+                    then
+                        return quest:progressEvent(665)
+                    end
+                end,
+            },
+            ['Rouva'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:hasKeyItem(xi.ki.GOLDSMITHING_ORDER) then
+                        return quest:event(664)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [665] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        quest:setVar(player, 'finalCS', 1)
+                        player:delKeyItem(xi.ki.MYTHRIL_HEARTS)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.BASTOK_MARKETS] =
+        {
+            ['Michea'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                    npcUtil.tradeHasExactly(trade, xi.item.MYTHRIL_INGOT) and
+                        quest:getVar(player, 'Prog') == 1
+                    then
+                        return quest:progressEvent(317)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if player:hasKeyItem(xi.ki.GOLDSMITHING_ORDER) then
+                        return quest:progressEvent(315)
+                    elseif questProgress == 1 then
+                        return quest:event(316)
+                    elseif
+                        questProgress == 2 and
+                        not player:needToZone()
+                    then
+                        return quest:progressEvent(318)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [315] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.GOLDSMITHING_ORDER)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+
+                [317] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    quest:setVar(player, 'Prog', 2)
+                    player:needToZone(true)
```

Comment:
Convention is to use the quest's setMustZone function.  needToZone is too generic.